server/galaxyls/tests/files/complex_inputs_01.xml

```diff
@@ -0,0 +1,56 @@
+<tool id="complex_inputs_01" name="Complex inputs">
+    <inputs>
+        <conditional name="c1">
+            <param name="c1_action" type="select">
+                <option value="a1" selected="True">A 1</option>
+                <option value="a2">A 2</option>
+                <option value="a3">A 3</option>
+            </param>
+            <when value="a1">
+                <param name="c1_a1_p1" type="text"/>
+                <section name="sec1" title="C1 A1 Section" expanded="true">
+                    <param name="number" value="1" type="integer"/>
+                </section>
+            </when>
+            <when value="a2">
+                <param name="c1_a2_p1" type="text"/>
+                <conditional name="c2">
+                    <param name="c2_action" type="select">
+                        <option value="c2a1" selected="True">A 1</option>
+                        <option value="c2a2">A 2</option>
+                    </param>
+                    <when value="c2a1">
+                        <param name="c2_a1_p1" type="text"/>
+                        <section name="sec_c2a1" title="C2 A1 Section" expanded="true">
+                            <section name="sub_sec_c2a1" title="C2 A1 Section" expanded="true">
+                                <param name="sub_p" value="3.2" type="float"/>
```

Comment:
If you add here a int, float, boolean and a data param ... I guess this is as complex as we can get for the time being :)

Reply:
I will add them for the sake of completion, but I think the result will be the same. I mean, the only *types* of `param` that actually have an effect in the generated test snippet are `boolean` and `select`.

Reply:
Is that intended, or something you plan to implement later?

Reply:
It is just that I don't know how they are translated to the test `param`, apart from taking the `name` and `value` as default, which all params do at the moment. Is there anything else those types should generate on the test param?

Reply:
Oh, please ask :)

Its always `name` and `value`. In case of boolean the only valid values are `true` and `false`.

Reply:
Here is a list of a bunch of test tools if you wan to get inspiration: https://github.com/galaxyproject/galaxy/tree/dev/test/functional/tools